src/sidebar/components/test/annotation-editor-test.js

```diff
@@ -0,0 +1,313 @@
+import { $imports } from '../annotation-editor';
+
+import { checkAccessibility } from '../../../test-util/accessibility';
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+
+describe('AnnotationEditor', () => {
+  let fakeApplyTheme;
+  let fakeAnnotationsService;
+  let fakeTagsService;
+  let fakeSettings;
+  let fakeToastMessenger;
+
+  let fakeStore;
+
+  function createComponent(props = {}) {
+    return mount(
+      <AnnotationEditor
+        annotation={fixtures.defaultAnnotation()}
+        annotationsService={fakeAnnotationsService}
+        settings={fakeSettings}
+        tags={fakeTagsService}
+        toastMessenger={fakeToastMessenger}
+        {...props}
+      />
+    );
+  }
+
+  beforeEach(() => {
+    fakeApplyTheme = sinon.stub();
+    fakeAnnotationsService = {
+      save: sinon.stub(),
+    };
+    fakeSettings = {};
+    fakeTagsService = {
+      store: sinon.stub(),
+    };
+    fakeToastMessenger = {
+      error: sinon.stub(),
+    };
+
+    fakeStore = {
+      createDraft: sinon.stub(),
+      getDraft: sinon.stub().returns(fixtures.defaultDraft()),
+      getGroup: sinon.stub(),
+    };
+
+    $imports.$mock(mockImportedComponents());
+    $imports.$mock({
+      '../store/use-store': callback => callback(fakeStore),
+      '../util/theme': { applyTheme: fakeApplyTheme },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('is an empty component if there is no draft', () => {
+    fakeStore.getDraft.returns(null);
+
+    const wrapper = createComponent();
+
+    assert.isEmpty(wrapper);
+  });
+
+  describe('markdown content editor', () => {
+    it('applies theme', () => {
+      const textStyle = { fontFamily: 'serif' };
+      fakeApplyTheme
+        .withArgs(['annotationFontFamily'], fakeSettings)
+        .returns(textStyle);
+
+      const wrapper = createComponent();
+      assert.deepEqual(
+        wrapper.find('MarkdownEditor').prop('textStyle'),
+        textStyle
+      );
+    });
+
+    it('updates draft text on edit callback', () => {
+      const wrapper = createComponent();
+      const editor = wrapper.find('MarkdownEditor');
+
+      act(() => {
+        editor.props().onEditText({ text: 'updated text' });
+      });
+
+      const call = fakeStore.createDraft.getCall(0);
+      assert.calledOnce(fakeStore.createDraft);
+      assert.equal(call.args[1].text, 'updated text');
+    });
+
+    describe('tag editing', () => {
+      it('adds tag when add callback invoked', () => {
+        const wrapper = createComponent();
+        wrapper.find('TagEditor').props().onAddTag('newTag');
+
+        const storeCall = fakeTagsService.store.getCall(0);
+        const draftCall = fakeStore.createDraft.getCall(0);
+
+        assert.deepEqual(storeCall.args[0], [{ text: 'newTag' }]);
+        assert.deepEqual(draftCall.args[1].tags, ['newTag']);
+      });
+
+      it('trims tag when add callback invoked', () => {
```

Comment:
I wonder if the trimming should be a responsibility of the tag editor or the annotation editor. If we wanted to re-use the tag editor elsewhere it might be easier to have it handle this.